manifests/init.pp

```diff
@@ -101,53 +101,35 @@
-  $namedconf_template    = $::dns::params::namedconf_template,
-  $acls                  = $::dns::params::acls,
-  $optionsconf_template  = $::dns::params::optionsconf_template,
-  $controls              = $::dns::params::controls,
-  $service_ensure        = $::dns::params::service_ensure,
-  $service_enable        = $::dns::params::service_enable,
-  $additional_options    = $::dns::params::additional_options,
-  $additional_directives = $::dns::params::additional_directives,
+  Stdlib::Absolutepath $namedconf_path                            = $::dns::params::namedconf_path,
+  Stdlib::Absolutepath $dnsdir                                    = $::dns::params::dnsdir,
+  String $dns_server_package                                      = $::dns::params::dns_server_package,
+  Stdlib::Absolutepath $rndckeypath                               = $::dns::params::rndckeypath,
+  Stdlib::Absolutepath $optionspath                               = $::dns::params::optionspath,
+  Stdlib::Absolutepath $publicviewpath                            = $::dns::params::publicviewpath,
+  Stdlib::Absolutepath $vardir                                    = $::dns::params::vardir,
+  String $namedservicename                                        = $::dns::params::namedservicename,
+  Stdlib::Absolutepath $zonefilepath                              = $::dns::params::zonefilepath,
+  Optional[Stdlib::Absolutepath] $localzonepath                   = $::dns::params::localzonepath,
+  Optional[Enum['only', 'first']] $forward                        = $::dns::params::forward,
+  Array[String] $forwarders                                       = $::dns::params::forwarders,
+  Optional[Variant[String, Boolean]] $listen_on_v6                = $::dns::params::listen_on_v6,
+  Enum['yes', 'no'] $recursion                                    = $::dns::params::recursion,
+  Array[String] $allow_recursion                                  = $::dns::params::allow_recursion,
+  Array[String] $allow_query                                      = $::dns::params::allow_query,
+  Enum[yes, no] $empty_zones_enable                               = $::dns::params::empty_zones_enable,
+  Optional[Enum['yes', 'no', 'explicit']] $dns_notify             = $::dns::params::dns_notify,
+  Enum['yes', 'no'] $dnssec_enable                                = $::dns::params::dnssec_enable,
+  Enum['yes', 'no', 'auto'] $dnssec_validation                    = $::dns::params::dnssec_validation,
+  String $namedconf_template                                      = $::dns::params::namedconf_template,
+  Hash[String, Array[String]] $acls                               = $::dns::params::acls,
+  String $optionsconf_template                                    = $::dns::params::optionsconf_template,
+  Hash[String, Hash[String, Data]] $controls                      = $::dns::params::controls,
+  Enum['running', 'true', 'stopped', 'false'] $service_ensure     = $::dns::params::service_ensure,
```

Comment:
I'd like to go with the second variant, as per https://docs.puppet.com/puppet/latest/types/service.html#service-attribute-ensure booleans are still valid.